.github/workflows/test.yml

```diff
@@ -0,0 +1,42 @@
+# This is a basic workflow to help you get started with Actions
+
+name: Unit tests
+
+# Controls when the workflow will run
+on:
+  # Triggers the workflow on push or pull request events but only for the master branch
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Set up Python 3.10
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
```

Comment:
didn't we moved to python 3.12 ?

Reply:
I'm not aware of it and `pyproject.toml` has 3.10 specified

Reply:
> Production now runs 3.12 yes, toml currently specifies minimum as 3.10. We should probably test with 3.12 and maybe bump that version as well.

Not probably, we should strive for that code and tests in the repo are done with the same environment as production.

Not doing so, means we can easily break production in some code path, that we won't see, even if there is a test that covers it.

Reply:
ok, I updated it, with minimal dependency changes.